test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+
+}
+
+module AnotherModule {
+  var a = false;
+}
+
+module ThirdModule {
+  var b = -13.0;
+}
+
+
+module Conflict {
+  /* This variable shares a name with a symbol in modToUse. */
+  var bar = 5;
+
+  var other = 5.0 + 3i;
+
+  var another = false;
+}
+
+
+module DifferentArguments {
+  /* This function shares a name with a function in modToUse, but requires
+     different arguments */
+  proc baz(x) {
+    return x - 2;
+  }
+}
+
+module MainModule {
+  proc main() {
+    writeln("Access from outside a module");
+
+    /* If a module is not the main module for a program, it is desirable for its
+       contents to be accessible to external modules.  There are several
+       strategies for accomplishing this:
+
+       First, a symbol can be referenced explicitly - this is done using the
+       module name and a separating '.' as a prefix to the name of the symbol
+       desired.
+    */
+    var thriceFoo = 3 * modToUse.foo; // should be 36
+    writeln(thriceFoo);
+
+
+    {
+      /* If many of the module's symbols are desired, or the same symbol is
+         desired multiple times, then it is best to utilize what is known as a
+         "use statement".
+      */
+      use modToUse;
+
+      /* Use statements can be inserted at any lexical scope that contains
+         executable code.
+
+         A use statement makes all of the module's visible symbols available to
+         the scope which contains the use statement.  These symbols may then be
+         accessed without the module name prefix.
+
+         In this case, bazBarFoo should store the result of calling modToUse.baz
+         on modToUse.bar and modToUse.foo, which is in this case '28'.
+      */
+      var bazBarFoo = baz(bar, foo);
+      writeln(bazBarFoo);
+    }
+
+
+
+    /* Since the following line doesn't live within a scope that contains a
+       'use' of 'modToUse', it would generate an error if uncommented.  This is
+       because 'foo' cannot be directly referenced, and is not qualified with a
+       module name.
+    */
+    // var twiceFoo = 2 * foo;
+
+
+
+    {
+      var bazBarFoo = baz(bar, foo);
+
+      /* Use statements apply to the entire scope in which they are defined.
+         Even if the use statement occurs after code which would directly refer
+         to its symbols, these references are still valid.  This is similar to
+         other Chapel forms of introducing symbols - for instance, class
+         declaration order does not prevent the first declared class from
+         referring to the second.
+
+         Thus, as in an earlier example, bazBarFoo should store the result of
+         calling modToUse.baz on modToUse.bar and modToUse.foo, which is in this
+         case '28'.
+      */
+      use modToUse;
+
+      writeln(bazBarFoo);
+    }
+
+
+    {
+      /* The symbols provided by a use statement are only considered when the
+         name in question cannot be resolved directly within the local scope.
```

Comment:
This only applies to non-functions, right?  This is why I was thinking it'd be nice to refer to function overloading early and get it out of the way, and then to state that the rest of the primer would only deal with non-functions.